sw/device/lib/testing/clkmgr_testutils.c

```diff
@@ -0,0 +1,36 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/clkmgr_testutils.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/testing/check.h"
+
+bool clkmgr_testutils_get_trans_clock_status(
+    dif_clkmgr_t *clkmgr, dif_clkmgr_params_t params,
```

Comment:
Please leave the params out of this function's args, and just use it in the implementation for bounds checks.

Reply:
The DIF `dif_clkmgr_hintable_clock_get_enabled` needs the `params` arg. The goal of this PR is to not fix the DIFs (the SW team is super passionate about how the DIFs must be written), but to introduce some reusable code. How about we address this in a future refactor? I would think this will prompt a more elaborate discussion with the SW team (who will have way stronger opinions than us =))

Reply:
My point is that params could be just supplied to `dif_clkmgr_hintable_clock_get_enabled` inside the body of `clkmgr_testutils_get_trans_clock_status`. On the other hand, I can't see where these params are defined... weird.

Reply:
Yes, currently you need to manually construct it (as you mention is done in clkmgr_smoketest.c). It is understandable that this is annoying and hurts usability. I am working on some proposals (and a matching PR) to address this (and the larger problem at heart described in #8409), so that should be solved shortly. Sorry for the temporary inconvenience.